bigtable-hbase-dataflow/src/test/java/com/google/cloud/bigtable/dataflow/CloudBigtableIOIntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+public class CloudBigtableIOIntegrationTest {
+  private static final String BIGTABLE_PROJECT_KEY = "google.bigtable.project.id";
+  private static final String BIGTABLE_ZONE_KEY = "google.bigtable.zone.name";
+  private static final String BIGTABLE_CLUSTER_KEY = "google.bigtable.cluster.name";
+
+  public static final byte[] COLUMN_FAMILY = Bytes.toBytes("test_family");
+  public static final byte[] QUALIFIER1 = Bytes.toBytes("qualifier1");
+
+  private static String projectId = System.getProperty(BIGTABLE_PROJECT_KEY);
+  private static String zoneId = System.getProperty(BIGTABLE_ZONE_KEY);
+  private static String clusterId = System.getProperty(BIGTABLE_CLUSTER_KEY);
+
+  public static TableName newTestTableName() {
+    return TableName.valueOf("test_table-" + UUID.randomUUID().toString());
+  }
+
+  private static BigtableConnection connection;
+  private static CloudBigtableConfiguration config;
+
+  @SuppressWarnings("rawtypes")
+  @BeforeClass
+  public static void setup() throws IOException {
+    config =
+        new CloudBigtableConfiguration.Builder()
+            .withProjectId(projectId)
+            .withZoneId(zoneId)
+            .withClusterId(clusterId)
+            .build();
+    connection = new BigtableConnection(config.toHBaseConfig());
+  }
+
+  @AfterClass
+  public static void shutdown() {
+    connection.close();
+  }
+
+  @Test
+  public void testEstimatedAndSplitForSmallTable() throws Exception {
+    TableName tableName = newTestTableName();
+    try (Admin admin = connection.getAdmin()) {
+      admin.createTable(
+          new HTableDescriptor(tableName).addFamily(new HColumnDescriptor(COLUMN_FAMILY)));
+      try (Table table = connection.getTable(tableName)) {
+        table.put(Arrays.asList(
+          new Put(Bytes.toBytes("row1")).addColumn(COLUMN_FAMILY, QUALIFIER1, Bytes.toBytes("1")),
+          new Put(Bytes.toBytes("row2")).addColumn(COLUMN_FAMILY, QUALIFIER1, Bytes.toBytes("2"))));
+      }
+
+      List<SampleRowKeysResponse> sampleRowKeys = getSampleKeys(tableName);
+
+      try {
+        CloudBigtableIO.Source source =
+            new CloudBigtableIO.Source(new CloudBigtableScanConfiguration(
+                projectId, zoneId, clusterId, tableName.getQualifierAsString(), new Scan()));
+        long estimatedSizeBytes = source.getEstimatedSizeBytes(null);
+        SampleRowKeysResponse lastSample = sampleRowKeys.get(sampleRowKeys.size() - 1);
+        Assert.assertEquals(lastSample.getOffsetBytes(), estimatedSizeBytes);
+
+        List<SourceWithKeys> bundles = source.splitIntoBundles(estimatedSizeBytes / 2 + 1, null);
+        // This will be a small table with no splits, so we return HConstants.EMPTY_START_ROW
+        // which can't be split.
+        Assert.assertEquals(sampleRowKeys.size() * 2 - 1, bundles.size());
```

Comment:
Might be clearer if you actually check that sampleRowKeys.size()  and bundles.size() are 1. Also a check on the  estimatedSizeBytes will make things clearer.